## Codificacion de estos datos

No todos nuestros datos contaran con un sencillo "nan" en nuestras filas, en ocaciones contaremos con otras formas de representar los datos faltantes que no reconocera pandas y lo tomara como un dato. Dato Erroneo :(

In [1]:
%run 0.1-dataset.ipynb
%run pandas-API.ipynb

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload
The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload
The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [25]:
common_na_strings = [
    "missing",
    "NA",
    "N A",
    "N/A",
    "#N/A",
    "NA ",
    " NA",
    "N /A",
    "N / A",
    " N / A",
    "N / A ",
    "na",
    "n a",
    "n/a",
    "na ",
    " na",
    "n /a",
    "n / a",
    " a / a",
    "n / a ",
    "NULL",
    "null",
    "",
    "?",
    "*",
    ".",
]

In [28]:
#Valores faltantes: 
datos.missing.number_missing()

63731

In [32]:
datos.dtypes

estoy_trabajando_en                                          object
donde_estas_trabajando                                       object
dedicacion                                                   object
tipo_de_contrato                                             object
ultimo_salario_mensual_o_retiro_bruto_en_tu_moneda_local    float64
ultimo_salario_mensual_o_retiro_neto_en_tu_moneda_local      object
pagos_en_dolares                                             object
dolarizado_cual_fue_el_ultimo_valor                          object
pagos_en_criptomonedas                                       object
que_porcentaje_o_monto_recibis_en_criptomonedas              object
recibis_algun_tipo_de_bono                                   object
a_que_esta_atado_el_bono                                     object
actualizaciones_de_tus_ingresos_laborales_durante_2022       object
de_que_fue_el_ajuste_total_acumulado                        float64
en_que_mes_fue_el_ultimo_ajuste                 

 Sustituimos globalmente para todas nuestras variables, afortunadamente no habian muchos datos pasando de tener 63731 datos faltantes a 63746.

In [29]:
#SUSTITUCION GLOBAL
datos_sustituidos = (
    datos
    .replace(
        to_replace= common_na_strings,
        value = np.nan
    )
)

In [31]:
datos_sustituidos.missing.number_missing()

63746

### Datos Faltantes implicitos
 Despues de experimentar con nuestro data set y realizar algunas pruebas, no se encontro datos implicitos que se puedan mostrar.

Un valor faltante implícito indica que el valor faltante debería estar incluido en el conjunto de datos del análisis, sin que éste lo diga o lo especifique. Por lo general, son valores que podemos encontrar al pivotar nuestros datos o contabilizar el número de apariciones de combinaciones de las variables de estudio.
